### PR TITLE
Static asset cleanup

### PR DESCRIPTION
**Pull request by happythenewsad**
## WHAT
- remove all Asset Pipeline precompiled assets 

## WHY
- we can port them to Vite, to simplify our asset pipeline 

## HOW
- update precompile directives in all 3 rails envs 
- remove all asset references from the assets.rb initializer 
- move precompiled assets to `entrypoints/snippets` (to differentiate them from regular entrypoints)
- update all erb tags to use the new locations of these assets 

### Notion Card Links
https://www.notion.so/quill/ea241f31a84c4b79ad56e83e99f7ee93?v=388007d5669c4347b4215e24770ebbcd&p=99c3403ad05943afa743c6795e922a03&pm=s

### What have you done to QA this feature?
1. In staging, send a daily stats email to myself, before and after the patch, and compare side by side 

```ruby 
UserMailer.class_eval do 
  def daily_stats_email(date_string)
    # Sidekiq converts variables to strings, so we recreate the Time object with the date string
    date_object = Time.zone.parse(date_string)
    start_time = date_object.beginning_of_day
    end_time = date_object.end_of_day
    subject_date = date_object.strftime('%m/%d/%Y')

    teacher_count = 999
    new_premium_accounts = 999
    conversion_rate = new_premium_accounts/teacher_count.to_f

    @current_date = date_object.strftime("%A, %B %d")
    @daily_active_teachers = 999
    @daily_active_students = 999 
    @new_teacher_signups = 999
    @new_student_signups = 999
    @classrooms_created = 999
    @activities_assigned = 999
    # Sentences written is quantified by number of activities completed multiplied by 10 because
    # there are an average of 10 sentences per activity.
    @sentences_written = 99
    @diagnostics_completed = 99
    @teacher_conversion_rate = 1
    @support_tickets_resolved = 0 
    @satismeter_nps_data = {'nps' => nil, 'respondents' => [0,1,2,3]}
    @satismeter_comment_data = []

    mail to: "pkong@quill.org", subject: "Quill Daily Analytics - #{subject_date}"
  end
end
```
Screenshot 2024-04-12 at 2 26 21 PM

2. In staging, run this hacked version of the `AdminUsageSnapshotEmailWorker`, and check the PDF for visual consistency:

```ruby
  class PDFTester
    class CloudUploadError < StandardError; end

    PDF_TEMPLATE = 'pdfs/admin_usage_snapshot_report'

    # The response-content-disposition triggers browser file download instead of screen rendering
    RESPONSE_CONTENT_DISPOSITION= { "response-content-disposition" => "attachment;" }

    attr_reader :pdf_subscription

    delegate :admin_report_filter_selection, :user, to: :pdf_subscription

    def initialize(pdf_subscription_id)
      @pdf_subscription_id = pdf_subscription_id
    end

    def perform
      @pdf_subscription = ::PdfSubscription.find(@pdf_subscription_id)

      return unless user.school_or_district_premium?

      upload_pdf
      puts download_url

    end

    private def data = ::Pdfs::AdminUsageSnapshotReports::DataAggregator.run(admin_report_filter_selection)

    private def upload_pdf
      ::Pdfs::FileBuilder.run(data:, template: PDF_TEMPLATE) do |tempfile|
        raise CloudUploadError, "Unable to upload PDF for user #{user_id}" unless uploader.store!(tempfile)
      end
    end

    private def uploader = @uploader ||= ::Pdfs::AdminUsageSnapshotReportUploader.new(user_id:)
    private def download_url = uploader.url(query: RESPONSE_CONTENT_DISPOSITION)
    private def user_id = user.id
  end

PDFTester.new(22).perform
```
Screenshot 2024-04-12 at 2 25 50 PM

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  yes
Have you deployed to Staging? | yes
Self-Review: Have you done an initial self-review of the code below on Github? |

### CODE REVIEW

File: services/QuillLMS/config/environments/production.rb [hunk 1]
```diff
@@ -66,13 +66,8 @@
...
   # Precompile additional assets.
   # application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
-  config.assets.precompile += %w(
-                                 cms.js
-                                 scorebook/*
-                                 scorebook/**/*
-                                 sign_up_email.css
-                                 admin_usage_snapshot_report_pdf.scss
-                                 )
+  # config.assets.precompile += []
```
Comment (happythenewsad) [inline services/QuillLMS/config/environments/production.rb:69]: This comment may be helpful to people hunting for the `precompile` directive.

**Comment by happythenewsad** [comment]
I have no expectation that folks deploy my code while I'm on vacation, but if you want to, thanks in advance :-)

File: services/QuillLMS/config/environments/production.rb (hunk 1 above)
Comment (brendanshean) [reply services/QuillLMS/config/environments/production.rb:69]: Great idea!

**Review by brendanshean — APPROVED** [review]
I really appreciate the clean-up here.  Thank you!

File: services/QuillLMS/spec/mailers/teacher_notifications/rollup_mailer_spec.rb [hunk 2]
```diff
@@ -17,6 +17,7 @@ module TeacherNotifications
     it { expect(mail.from).to eq(['hello@quill.org']) }
 
     it 'should include the user notification frequency value in the subject line' do
+      #binding.pry
```
Comment (emilia-friedberg) [inline services/QuillLMS/spec/mailers/teacher_notifications/rollup_mailer_spec.rb]: Can be deleted.

**Review by emilia-friedberg — APPROVED** [review]
LGTM! Good solution here.

**Pull request merged**